packages/js/src/dashboard/store/alert-center.js

```diff
@@ -0,0 +1,116 @@
+/* global ajaxurl */
+import { createSlice, createSelector } from "@reduxjs/toolkit";
+import { get } from "lodash";
+import { ASYNC_ACTION_NAMES, ASYNC_ACTION_STATUS } from "../../shared-admin/constants";
+
+export const ALERT_CENTER_NAME = "alertCenter";
+
+const TOGGLE_ALERT_VISIBILITY = "toggleAlertVisibility";
+
+/**
+ * @param {string} id The id of the alert.
+ * @param {string} nonce The nonce of the alert.
+ * @param {boolean} hidden The hidden state of the alert.
+ * @returns {Object} Success or error action object.
+ */
+export function* toggleAlertStatus( id, nonce, hidden = false ) {
+	yield{ type: `${ TOGGLE_ALERT_VISIBILITY }/${ ASYNC_ACTION_NAMES.request }` };
+	try {
+		yield{ type: TOGGLE_ALERT_VISIBILITY,
+			payload: {
+				id,
+				nonce,
+				hidden,
+			},
+		    };
+		return { type: `${ TOGGLE_ALERT_VISIBILITY }/${ ASYNC_ACTION_NAMES.success }`, payload: { id } };
+	} catch ( error ) {
+		return { type: `${ TOGGLE_ALERT_VISIBILITY }/${ ASYNC_ACTION_NAMES.error }`, payload: error };
+	}
+}
+
+/**
+ * Toggle alert visability.
+ *
+ * @param {object} state The state.
+ * @param {string} id The id of the alert to hide.
+ *
+ * @returns {void}
+ */
+const toggleAlert = ( state, id ) => {
+	state.alerts = state.alerts.map( ( alert ) => {
+		if ( alert.id === id ) {
+			return { ...alert, dismissed: ! alert.dismissed };
+		}
+		return alert;
+	} );
+};
+
+const slice = createSlice( {
+	name: ALERT_CENTER_NAME,
+	initialState: { alerts: [] },
+	reducers: {
+		toggleAlert,
+	},
+	extraReducers: ( builder ) => {
+		builder.addCase( `${ TOGGLE_ALERT_VISIBILITY }/${ ASYNC_ACTION_STATUS.success }`, ( state, { payload: { id } } ) => {
+			slice.caseReducers.toggleAlert( state, id );
+		} );
+	},
+} );
+
+/**
+ * @returns {Object} The initial state.
+ */
+export const createInitialAlertCenterState = slice.getInitialState;
+
+/**
+ * Base selector to get the alerts from the state.
+ *
+ * @param {object} state The state.
+ * @returns {array} The alerts.
+ */
+const selectAlerts = ( state ) => get( state, "alertCenter.alerts", [] );
+
+export const alertCenterSelectors = {
```

Comment:
Pretty cool.
It could even be less code with something like. But that might be over-engineering it 😬 

```js
const selectFilteredAlerts = createSelector(
	[
		selectAlerts,
		( state, type ) => type,
		( state, type, dismissed ) => dismissed,
	],
	( alerts, type, dismissed ) => alerts.filter( ( alert ) => type === alert.type && dismissed === alert.dismissed )
);
```

Though if exposing this, it would open up the code to just use these `type` and `dismissed` (visibilityStatus?) as props. Not relying on specific names any longer.